src/main/java/pl/javastart/task/TournamentStats.java

```diff
@@ -1,12 +1,103 @@
 package pl.javastart.task;
 
-import java.util.Scanner;
+import java.util.*;
 
 public class TournamentStats {
 
+    private static final int SORT_BY_FIRST_NAME = 1;
+    private static final int SORT_BY_LAST_NAME = 2;
+    private static final int SORT_BY_RESULT = 3;
+    private static final int SORTING_BY_REVERSE = 2;
+    private static final int SORT_ASCENDING = 1;
+    private static boolean reverseList;
+
     void run(Scanner scanner) {
-        // tutaj dodaj swoje rozwiązanie
-        // użyj przekazanego scannera do wczytywania wartości
+        TournamentStatsWriter tournamentStatsWriter = new TournamentStatsWriter();
+        List<TournamentPlayer> tournamentPlayers = listOfPersonsMaker(scanner);
+        sortListOfPlayers(scanner, tournamentStatsWriter, tournamentPlayers);
+    }
+
+    private static void sortListOfPlayers(Scanner scanner, TournamentStatsWriter tournamentStatsWriter, List<TournamentPlayer> tournamentPlayers) {
```

Comment:
Ta metoda tworzy komparator, sortuje i zapisuje w pliku, nazwa nie tego nie oddaje. Podziel ją własnie na te etapy:
tworzenie komparatora
posortowanie listy
zapis do pliku
Najlepiej, żeby to było widoczne w metodzie run